tests/ui-toml/bool_comparison_equal/bool_comparison_equal.stderr

```diff
@@ -0,0 +1,52 @@
+error: inequality checks against true can be replaced with a comparison against `false`
+  --> $DIR/bool_comparison_equal.rs:7:13
+   |
+LL |     let _ = x != true;
+   |             ^^^^^^^^^ help: try simplifying it as shown: `x == false`
+   |
+   = note: `-D clippy::bool-comparison` implied by `-D warnings`
+
+error: inequality checks against true can be replaced with a comparison against `false`
+  --> $DIR/bool_comparison_equal.rs:8:13
+   |
+LL |     let _ = true != x;
+   |             ^^^^^^^^^ help: try simplifying it as shown: `x == false`
+
+error: less than comparison against true can be replaced with a comparison against `false`
+  --> $DIR/bool_comparison_equal.rs:9:13
+   |
+LL |     let _ = x < true;
+   |             ^^^^^^^^ help: try simplifying it as shown: `x == false`
+
+error: less than comparison against true can be replaced with a comparison against `false`
+  --> $DIR/bool_comparison_equal.rs:10:13
+   |
+LL |     let _ = true > x;
+   |             ^^^^^^^^ help: try simplifying it as shown: `x == false`
+
+error: order comparisons between booleans can be simplified
+  --> $DIR/bool_comparison_equal.rs:13:13
+   |
+LL |     let _ = x < y;
+   |             ^^^^^ help: try simplifying it as shown: `x == false & y`
```

Comment:
The `&` operator is semanticly equivalent, so i think it's good to have it as the main suggestion. Especially if the suggestion might become machine applicable at some point. However, it might also be good to include a help comment, suggesting the usage of `&&`. :)

Reply:
The only thing that differs this diagnostic from the current status-quo is `x == false` instead of `!x`.

https://github.com/rust-lang/rust-clippy/blob/dc17e7317b7bd6d585abafa4d927dcf876dc4d18/tests/ui/bool_comparison.stderr#L75-L79

So if possible I prefer to improve messages in a following PR.

Reply:
> The only thing that differs this diagnostic from the current status-quo is `x == false` instead of `!x`.

Exactly, I think it would be better to suggest `!x` directly, as it would otherwise be linted in the next clippy run

> So if possible I prefer to improve messages in a following PR.

Sure that works for me :)